packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1478,4 +1482,90 @@ export default abstract class SqlIntegration
 
     throw new Error(`Missing identifier join table for '${id1}' and '${id2}'.`);
   }
+
+  async getPostgresInformationSchema(
+    dataSourceType: DataSourceType
+  ): Promise<InformationSchema[]> {
+    const sql = `SELECT
+        table_name,
+        column_name,
+        data_type,
+        table_catalog,
+        table_schema
+      FROM
+        information_schema.columns
+      WHERE
+        table_schema
+      NOT IN ('pg_catalog', 'information_schema', 'pg_toast')
+      ORDER BY table_name;`;
+
+    const results = await this.runQuery(sql);
+
+    return results.length
+      ? formatInformationSchema(
+          results as RawInformationSchema[],
+          dataSourceType
+        )
+      : [];
+  }
+
+  async getBigQueryInformationSchema(
+    dataSourceType: DataSourceType,
+    projectId: string
+  ): Promise<InformationSchema[]> {
+    if (!projectId) {
+      return [];
+    }
+
+    const datasets = await this.runQuery(
+      `SELECT * FROM ${projectId}.INFORMATION_SCHEMA.SCHEMATA;`
+    );
+
+    const combinedResults: RawInformationSchema[] = [];
+
+    const query: string[] = [];
+
+    for (const dataset of datasets) {
+      query.push(`SELECT
+        "${projectId}" as table_catalog,
+        table_name,
+        column_name,
+        data_type,
+        table_schema
+      FROM
+        ${projectId}.${dataset.schema_name}.INFORMATION_SCHEMA.COLUMNS`);
+    }
+
+    let queryString = query.join(" UNION ALL ");
+
+    queryString = queryString + " ORDER BY table_name;";
+
+    const results = await this.runQuery(queryString);
+
+    for (const result of results) {
+      combinedResults.push(result);
+    }
+
+    return combinedResults.length
+      ? formatInformationSchema(combinedResults, dataSourceType)
+      : [];
+  }
+
+  async getInformationSchema(
```

Comment:
Correct me if I'm wrong @tinahollygb  but I think you were suggesting this to be defined as an [abstract method](https://www.tutorialsteacher.com/typescript/abstract-class), like so:

```
abstract getInformationSchema(dataSourceType: DSType, projectId: string): Promise<InformationSchema[] | null>;
```

And then the above two methods (bigquery, postgres, and so on) defined in their respected derived classes ... I agree that that pattern would make most sense and organize the code a little better.

Reply:
@bttf @tinahollygb Ah - I think I understand a bit better. I just pushed up this commit based on your comment, Adnan.

Let me know if that's closer to what you were suggesting.

Currently, I've defined `getInformationSchema` on all of the classes that extend `SQLIntegration` to make Typescript happy - I am currently exploring the following pattern below as an alternative.

```ts
function _notImplemented() : never {
     throw new Error("not Implemented")
}
```

Reply:
Yep, looks good. Throwing a 'not implemented' error is really just a style note - it's something done in other languages (I think .net actually supports an error of that type) - either way returning null is fine, too

Reply:
TypeScript and JavaScript do not have checked exceptions designed into the language, i.e. developers do not get any compiler help when a method throws, so I wouldn't do that in this case. Perhaps returning null or an empty array would be safer.